resources/js/beatmap-discussions/nominations.tsx

```diff
@@ -0,0 +1,680 @@
+        props: {
+          onClick: this.discussionLock,
+        },
+      };
+
+    return <BigButton {...buttonProps} disabled={this.xhr.discussionLock != null} text={trans('beatmaps.nominations.disqualify')} />;
+  }
+
+  private renderDiscussionLockMessage() {
+    if (!this.props.beatmapset.discussion_locked) return null;
+
+    const lockEvent = findLast(this.props.events, { type: 'discussion_lock' });
+
+    if (lockEvent == null || lockEvent.type !== 'discussion_lock') return null;
+
+    return (
+      <StringWithComponent
+        mappings={{
+          text: lockEvent.comment?.reason,
+        }}
+        pattern={trans('beatmapset_events.event.discussion_lock')}
+      />
+    );
+  }
+
+  private renderDisqualificationMessage() {
+    const showHype = this.props.beatmapset.can_be_hyped;
+    const disqualification = this.props.beatmapset.nominations.disqualification;
+    const mapIsQualified = this.props.beatmapset.status === 'qualified';
+
+    if (!showHype || mapIsQualified || disqualification == null) return null;
+
+    return <div>{this.renderResetReason(disqualification)}</div>;
+  }
+
+  private renderDisqualifyButton() {
+    const mapIsQualified = this.props.beatmapset.status === 'qualified';
+
+    if (!(mapIsQualified && this.userCanDisqualify)) return null;
+
+    return (
+      <BigButton
+        icon='fas fa-thumbs-down'
+        modifiers='warning'
+        props={{
+          onClick: this.focusNewDiscussionWithModeSwitch,
+        }}
+        text={trans('beatmaps.nominations.disqualify')}
+      />
+    );
+  }
+
+  private renderFeedbackButton() {
+    if (core.currentUser == null || this.userIsOwner || this.props.beatmapset.can_be_hyped || this.props.beatmapset.discussion_locked) {
+      return null;
+    }
+
+    return (
+      <BigButton
+        icon='fas fa-bullhorn'
+        props={{
+          onClick: this.focusNewDiscussionWithModeSwitch,
+        }}
+        text={trans('beatmaps.feedback.button')}
+      />
+    );
+  }
+
+  private renderHypeBar() {
+    if (!this.props.beatmapset.can_be_hyped || this.props.beatmapset.hype == null) return null;
+
+    const requiredHype = this.props.beatmapset.hype.required;
+    const hypeRaw = this.props.currentDiscussions.totalHype;
+    const hype = Math.min(requiredHype, hypeRaw);
+
+    return (
+      <div>
+        <div className={`${bn}__header`}>
+          <span className={`${bn}__title`}>{trans('beatmaps.hype.section_title')}</span>
+          <span>{`${hypeRaw} / ${requiredHype}`}</span>
+        </div>
+        <DiscreteBar
+          current={hype}
+          modifiers='beatmapset-hype'
+          total={requiredHype}
+        />
+      </div>
+    );
+  }
+
+  private renderHypeButton() {
+    if (!this.props.beatmapset.can_be_hyped || core.currentUser == null || this.userIsOwner) return null;
+
+    const userAlreadyHyped = find(this.props.currentDiscussions.byFilter.hype.generalAll, { user_id: core.currentUser.id }) != null;
```

Comment:
seems the same as `Array#find`. Although `Array#some` would be more to the point.